uat/custom-components/src/main/java/com/aws/greengrass/artifacts/ShadowComponent.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.artifacts;
+
+import com.aws.greengrass.utils.IPCTestUtils;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.aws.greengrass.GreengrassCoreIPC;
+import software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCClientV2;
+import software.amazon.awssdk.aws.greengrass.model.DeleteThingShadowRequest;
+import software.amazon.awssdk.aws.greengrass.model.DeleteThingShadowResponse;
+import software.amazon.awssdk.aws.greengrass.model.GetThingShadowRequest;
+import software.amazon.awssdk.aws.greengrass.model.GetThingShadowResponse;
+import software.amazon.awssdk.aws.greengrass.model.ListNamedShadowsForThingRequest;
+import software.amazon.awssdk.aws.greengrass.model.ListNamedShadowsForThingResponse;
+import software.amazon.awssdk.aws.greengrass.model.UpdateThingShadowRequest;
+import software.amazon.awssdk.aws.greengrass.model.UpdateThingShadowResponse;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+public class ShadowComponent implements Consumer<String[]> {
+    private String thingName;
+    private String operation;
+    private String shadowName;
+    private String shadowDocument;
+    private byte[] shadowDocumentPayload;
+    private Integer pageSize;
+    private String nextToken;
+    private int timeoutSeconds;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ShadowComponent.class);
+    private static final List<String> NAMED_SHADOWS_LIST = Arrays.asList("alpha", "bravo", "charlie", "delta");
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private GreengrassCoreIPC greengrassCoreIPCClient = null;
+
+    @Override
+    public void accept(String[] args) {
+        if (args.length != 7) {
+            LOGGER.error(
+                    "Wrong number of arguments. Expected arguments: <Operation> <ThingName> <ShadowName> <ShadowDocument> <PageSize> <NextToken> <Expected Document> <timeout>");
+            for (String arg : args) {
+                LOGGER.error(arg);
+            }
+            return;
+        }
+        parseArgs(args);
```

Comment:
Move the if inside of the parseArgs method

Reply:
Yeah now I'll have to throw exception if I have to exit from parseArgs..